### PR TITLE
civilian-borg cleaner bottle automatically refills

### PR DESCRIPTION
**Pull request by BenLubar**
[QOL][FEATURE]

## About the PR

The cyborg cleaner bottle now fills back up to 25% (1% per tick) automatically. Icon by Troasta.

## Why's this needed?

Convenience, mostly. This way, instead of getting 20 sprays per round, you get 20 sprays and if it's below 5 sprays it generates more cleaner for you. It's pretty awkward for cyborgs to make space cleaner using a chem dispenser, and they can't hold items, so most of the time after the initial space cleaner, you'd end up using plain water or just not using the spray bottle at all.

## Changelog
```
(u)BenLubar:
(*)Cyborg cleaner bottles now automatically synthesize space cleaner!
```

### CODE REVIEW

**Review by mordent-goonstation — COMMENTED** [review]
Have you checked this works? Unless I'm missing something, the spraybottle isn't added to the processing loop so `process` would never trigger.

File: code/obj/item/mops_cleaners.dm [hunk 1]
```diff
@@ -54,11 +54,16 @@ WET FLOOR SIGN
 	desc = "A cleaner bottle jury-rigged to synthesize space cleaner."
 	icon_state = "cleaner_robot"
 
+	New()
+		..()
+		processing_items.Add(src)
```
Comment (mordent-goonstation) [inline code/obj/item/mops_cleaners.dm:59]:
```suggestion
	New()
		..()
		processing_items.Add(src)

	disposing()
		..()
		processing_items.Remove(src)	
```
@pali6 Sanity check.

**Review by mordent-goonstation — CHANGES_REQUESTED** [review]

File: code/obj/item/mops_cleaners.dm (hunk 1 above)
Comment (BenLubar) [reply code/obj/item/mops_cleaners.dm:59]: there's no `disposing()` in https://github.com/goonstation/goonstation/blob/master/code/modules/chemistry/tools/robospray.dm, which is what I based this on

Comment (pali6) [reply code/obj/item/mops_cleaners.dm:59]: Yeah it should absolutely be removed in disposing.

Comment (pali6) [reply code/obj/item/mops_cleaners.dm:59]: Honestly I think I'd prefer to have the processing_items registration happen in `afterattack` where the spraying itself happens:
```
afterattack(atom/A, mob/user)
		. = ..()
		if (src.reagents.total_volume < 25)
      processing_items |= src
```
and then have
```
		if (src.reagents.total_volume < 25)
			src.reagents.add_reagent("cleaner", 1)
    else
      processing_items.Remove(src)
```
in `process`. That way you avoid it `processing` pointlessly when full.

**Review by pali6 — APPROVED** [review]

**Review by flrsh — APPROVED** [review]

**Pull request merged**